f5_openstack_agent/lbaasv2/drivers/bigip/test/test_vlan.py

```diff
@@ -0,0 +1,143 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from f5_openstack_agent.lbaasv2.drivers.bigip.network_helper import \
+    NetworkHelper
+
+from f5.bigip.tm.net.vlan import TagModeDisallowedForTMOSVersion
+
+import mock
+import pytest
+
+
+tagged_vlan_no_int = {'name': "test_valan",
+                      'partition': "Project_123456789",
+                      'tag': 1000}
+
+
+class TestVLANCreate(object):
+
+    @pytest.fixture
+    def bigip(self):
+        bigip = mock.MagicMock()
+        bigip.tm.net.vlans.vlan = mock.MagicMock()
+        bigip.tm.net.vlans.vlan.exists.return_value = True
+
+        return bigip
+
+    @pytest.fixture
+    def network_helper(self):
+        nh = NetworkHelper()
+        nh.get_route_domain_by_id = mock.MagicMock(return_value=0)
+
+        return nh
+
+    def test_create_vlan_no_name(self, bigip, network_helper):
+        tagged_vlan_no_name = {'partition': "Project_123456789"}
+
+        vlan = network_helper.create_vlan(bigip, tagged_vlan_no_name)
+
+        assert(vlan is None)
+
+    def test_create_vlan_exists(self, bigip, network_helper):
+        tagged_vlan_no_int = {'name': "test_vlan",
+                              'partition': "Project_123456789",
+                              'tag': 1000}
+
+        v = network_helper.create_vlan(bigip, tagged_vlan_no_int)
+
+        assert(v is not None)
+
+        bigip.tm.net.vlans.vlan.exists.assert_called_once_with(
+            name='test_vlan', partition='Project_123456789')
+        bigip.tm.net.vlans.vlan.load.assert_called_once_with(
+            name='test_vlan', partition='Project_123456789')
+
+    def test_create_vlan_no_int(self, bigip, network_helper):
+
+        tagged_vlan_no_int = {'name': "test_vlan",
+                              'partition': "Project_123456789",
+                              'tag': 1000}
+
+        bigip.tm.net.vlans.vlan.exists.return_value = False
+
+        v = network_helper.create_vlan(bigip, tagged_vlan_no_int)
+
+        assert(v is not None)
+
+        bigip.tm.net.vlans.vlan.create.assert_called_once_with(
+            name='test_vlan', partition='Project_123456789', tag=1000)
+
+    def test_create_vlan_with_untagged_int(self, bigip, network_helper):
+
+        vlan_with_untagged_int = {'name': "test_vlan",
+                                  'partition': "Project_123456789",
+                                  'interface': "1.3"}
+
+        bigip.tm.net.vlans.vlan.exists.return_value = False
+        bigip.tm.net.vlans.vlan.create.return_value = (
+            bigip.tm.net.vlans.vlan)
+
+        v = network_helper.create_vlan(bigip, vlan_with_untagged_int)
+
+        bigip.tm.net.vlans.vlan.create.assert_called_once_with(
+            name='test_vlan', partition='Project_123456789', tag=0)
+
+        v.interfaces_s.interfaces.create.assert_called_with(
```

Comment:
In the previous test, you can assert this function was not called on the v object.